ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -450,6 +465,22 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
         // workspace which references a non-default, non-archived CDR version.
         this.setState(fp.set(['workspace', 'cdrVersionId'], cdrResp.defaultCdrVersionId));
       }
+
+      if (serverConfigStore.getValue().enableBillingLockout) {
+        userApi().listBillingAccounts().then(response =>
+          this.setState({billingAccounts: response.billingAccounts})
+        );
+
+        if (this.isMode(WorkspaceEditMode.Edit)) {
+          getBillingAccountName(this.props.workspace.namespace).then(billingAccountName =>
```

Comment:
Some cases to make sure are handled here:

1. The user is an editor (not an owner). We expect this call to 403.
2. The user is an owner, but doesn't have access to the billing account (i.e. the workspace billing account is not present in the dropdown).

Reply:
It looks like the user is able to call `getBillingInfo` on the project when they're a writer but they are NOT able to update the billing info. Is this OK? Fixing the permissions so that they can't call `getBillingInfo` might be non trivial since I think that is managed on FC's side based on how they interpret the writer role.

From AoU's standpoint, they will not be able to update the workspace since only owners can edit Workspace details. We can also just remove the dropdown for writers if we do not want them to be able to see the billing info.

With regards to #2, the logic here got a bit tricky but I think I have something reasonable and there's enough comments that it should be readable. LMK what you think.

Reply:
Only owners should be able to change the billing information, so the dropdown should ideally just be disabled for non-owners.